### PR TITLE
Add more checks for trigger module

### PR DESCRIPTION
**Pull request by pankajastro**
Let's enable more checks for provider triggers modules.
In this PR, enable the following check
- Make sure the trigger class end with Trigger
- Trigger class if in providers.yaml then make sure it exists in the module
- Add a test provider_manager.trigger return non-zero class

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by eladkal** [comment]
is it part of https://github.com/apache/airflow/issues/29918 ?

**Comment by pankajastro** [comment]
> is it part of #29918 ?

some changes look close to the below point but not exactly. I think for the below point we want to add tests to make sure classes of trigger modules are present in provider.yaml but this PR doing otherwise. 
But I'm happy to include changes for this as well in this PR if we already do such a check for other classes like check if all `secrets-backends` module classes are present in provider.yaml. @eladkal  if we have tests for such as check can you please point me to the tests otherwise I'm thinking of doing it in a separate PR for trigger and others also WDYT?

> check trigge classes are present in provider.yaml in unit tests

File: scripts/in_container/run_provider_yaml_files_check.py [hunk 1]
```diff
@@ -306,6 +306,20 @@ def check_hook_classes(yaml_files: dict[str, dict]):
...
 
 
+def check_trigger_classes(yaml_files: dict[str, dict]):
+    print("Checking triggers classes belong to package, exist and are classes")
+    resource_type = "triggers"
+    for yaml_file_path, provider_data in yaml_files.items():
+        provider_package = pathlib.Path(yaml_file_path).parent.as_posix().replace("/", ".")
+        trigger_classes = []
+        for trigger_class in provider_data.get(resource_type, {}):
+            trigger_classes.extend(trigger_class["class-names"])
```
Comment (uranusjr) [inline scripts/in_container/run_provider_yaml_files_check.py:316]:
```suggestion
        trigger_classes = [
            name
            for trigger_class in provider_data.get(resource_type, {})
            for name in names in trigger_class["class-names"]
        ]
```

File: scripts/in_container/run_provider_yaml_files_check.py [hunk 2]
```diff
@@ -306,6 +306,20 @@ def check_hook_classes(yaml_files: dict[str, dict]):
...
+        if trigger_classes:
+            check_if_objects_exist_and_belong_to_package(
+                set(trigger_classes), provider_package, yaml_file_path, resource_type, ObjectType.CLASS
+            )
```
Comment (uranusjr) [inline scripts/in_container/run_provider_yaml_files_check.py]:
```suggestion
        trigger_classes = {
            name
            for trigger_class in provider_data.get(resource_type, {})
            for name in names in trigger_class["class-names"]
        }
        if trigger_classes:
            check_if_objects_exist_and_belong_to_package(
                trigger_classes, provider_package, yaml_file_path, resource_type, ObjectType.CLASS
            )
```

**Review by uranusjr — APPROVED** [review]

File: scripts/in_container/run_provider_yaml_files_check.py [hunk 3]
```diff
@@ -306,6 +306,22 @@ def check_hook_classes(yaml_files: dict[str, dict]):
...
 
+def check_trigger_classes(yaml_files: dict[str, dict]):
+    print("Checking triggers classes belong to package, exist and are classes")
+    resource_type = "triggers"
+    for yaml_file_path, provider_data in yaml_files.items():
+        provider_package = pathlib.Path(yaml_file_path).parent.as_posix().replace("/", ".")
+        trigger_classes = {
+            name
+            for trigger_class in provider_data.get(resource_type, {})
+            for name in names in trigger_class["class-names"]
```
Comment (pankajastro) [inline scripts/in_container/run_provider_yaml_files_check.py]:
```suggestion
            for name in trigger_class["class-names"]
```

**Review by eladkal — APPROVED** [review]
cc @potiuk

**Review by potiuk — APPROVED** [review]
Nice

**Pull request merged**